crates/napi/src/next_api/mod.rs

```diff
@@ -1,3 +1,16 @@
 pub mod endpoint;
 pub mod project;
 pub mod utils;
+
+use std::sync::{LazyLock, Mutex};
+
+// Unfortunately, napi-rs doesn't let us attach structured data to thrown errors. We store error
+// locations here so they can be separately requested by JS.
+pub static LAST_ERROR_LOCATION: LazyLock<Mutex<Option<String>>> =
```

Comment:
should we be concerned about dropping panics if multiple fire concurrently?  should this be a Vec instead? or a map keyed off the Error.message (gross).

maybe add a comment

Reply:
My understanding is that this will all happen synchronously (Rust panics, it sets the error location, JS receives the `Error`, then reads the location back). Will add a comment. cc @bgw

Reply:
@kdy1 Will and I discussed some alternatives

1. passing a custom error type from TS->Rust and we would construct that
    Pros: get some type safety from having a dedicated constructor/factory function
    Cons: subtyping error is relatively odd and it would complicate the normal calling conventions.
2. encode the structured data in json in the error message
    Pros: the data we want to pass is just a string and it isn't large so this is quite reasonable.
     Cons: these wouldn't be the only errors so we need to be defensive
3. don't use js Error types, embrace sum types and return a simple tuple of `[result, error]`

Reply:
I prefer 2, but I'm not sure if it's a good idea.